db/migrate/20151119070640_remove_dangling_entries.rb

```diff
@@ -0,0 +1,22 @@
+class RemoveDanglingEntries < ActiveRecord::Migration
+  def up
+    UserSnp.find_each do |us|
+      if Snp.find_by_name(us.snp_name).nil? || Genotype.find_by_id(us.id).nil?
+        UserSnp.destroy(us.id)
+      end
+    end
```

Comment:
It works due to some Rails magic - this is the command that is actually triggered for a UserSNP for the non-existing SNP-name "blu":

 `SQL (0.3ms)  DELETE FROM "user_snps" WHERE "user_snps"."genotype_id" = $1 AND "user_snps"."snp_name" = $2  [["genotype_id", 4], ["snp_name", "blu"]]`

But you're right, there's no ID, I assume that it can't find the ID, and then just hands the whole object to the destroy method 

I assume that we have some usersnps hanging around because sidekiq crashed/the parsing crashed, from what I've seen this used to happen, these *should* get deleted but let's just make sure